src/app/inventory/store/d2-item-factory.ts

```diff
@@ -76,23 +78,27 @@ export function processItems(
   owner: DimStore,
   items: DestinyItemComponent[],
   itemComponents: DestinyItemComponentSetOfint64,
-  uninstancedItemObjectives?: {
-    [key: number]: DestinyObjectiveProgress[];
-  },
-  profileRecords?: DestinyProfileRecordsComponent
+  uninstancedItemObjectives?: DestinyCharacterProgressionComponent['uninstancedItemObjectives'],
+  profileRecords?: DestinyProfileRecordsComponent,
+  uninstancedItemPerks?: DestinyCharacterProgressionComponent['uninstancedItemPerks']
 ): DimItem[] {
   const result: DimItem[] = [];
 
   for (const item of items) {
     let createdItem: DimItem | null = null;
+
+    const itemUninstancedObjectives = uninstancedItemObjectives?.[item.itemHash];
+    const itemUninstancedPerks = uninstancedItemPerks?.[item.itemHash]?.perks;
```

Comment:
it seemed weird to feed the entire global dict into `makeItem` (singular), just so that two functions deeper, it could narrow to only this item's relevant info. so we extract the single item's info beforehand.